protocol/message.go

```diff
@@ -39,10 +57,39 @@ type DirectoryProof struct {
 	AP    *m.AuthenticationPath
 	STR   *m.SignedTreeRoot
 	TB    *m.TemporaryBinding `json:",omitempty"`
-	Error ErrorCode           `json:",omitempty"`
+	Error ErrorCode
+}
+
+type DirectoryProofInEpoch struct {
+	Type  int
+	AP    *m.AuthenticationPath
+	STR   []*m.SignedTreeRoot
+	Error ErrorCode
+}
+
+type DirectoryProofs struct {
```

Comment:
I think we might be able to consolidate `DirectoryProofInEpoch` and `DirectoryProofs` and just have the client make sure that the length of the `aps` struct is of length 1 when doing a `KeyLookupInEpoch`.